BootloaderCommonPkg/Library/ShellLib/CmdLog.c

```diff
@@ -0,0 +1,104 @@
+/** @file
+  Shell command `log` to print the contents of the log buffer.
+
+  Copyright (c) 2018, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include <Library/ShellLib.h>
+#include <Library/DebugLib.h>
+#include <Library/HobLib.h>
+#include <Guid/LoaderFspInfoGuid.h>
+#include <Library/ConsoleInLib.h>
+#include <Library/ConsoleOutLib.h>
+#include <Library/BootloaderCommonLib.h>
+
+/**
+  Print the contents of the log buffer
+
+  @param[in]  Shell        shell instance
+  @param[in]  Argc         number of command line arguments
+  @param[in]  Argv         command line arguments
+
+  @retval EFI_SUCCESS
+
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+ShellCommandLogFunc (
+  IN SHELL  *Shell,
+  IN UINTN   Argc,
+  IN CHAR16 *Argv[]
+  );
+
+CONST SHELL_COMMAND ShellCommandLog = {
+  L"log",
```

Comment:
should we use "dmsg" so that it is similar to Linux ?

Reply:
Sure, that will probably be more familiar to users. I'll change it.

Reply:
Fixed